src/util/graph.h

```diff
@@ -177,10 +177,11 @@ class grapht
   nodest nodes;
 
 public:
-  node_indext add_node()
+  template <typename... arguments>
+  node_indext add_node(arguments &&... values)
   {
     node_indext no=nodes.size();
-    nodes.push_back(nodet());
```

Comment:
Would it be worth actually switching to `emplace_back(std::forward<arguments>(values)...)`?